frontend/app/src/components/history/events/HistoryEventTypeCounterparty.vue

```diff
@@ -1,52 +1,70 @@
 <script setup lang="ts">
-import {
-  type EthDepositEvent,
-  type EvmHistoryEvent
-} from '@/types/history/events';
-
-const props = defineProps<{
-  event: EvmHistoryEvent | EthDepositEvent;
-}>();
+const props = withDefaults(
+  defineProps<{
+    event: { counterparty?: string | null; address?: string | null };
+    text?: boolean;
+  }>(),
+  {
+    text: false
+  }
+);
 
 const { event } = toRefs(props);
 
 const { getEventCounterpartyData } = useHistoryEventMappings();
 
 const counterparty = getEventCounterpartyData(event);
 const imagePath = '/assets/images/protocols/';
+
+const [DefineImage, ReuseImage] = createReusableTemplate();
+const [DefineText, ReuseText] = createReusableTemplate();
 </script>
 
 <template>
   <div>
-    <VBadge v-if="counterparty || event.address" avatar overlap color="white">
-      <template #badge>
-        <VTooltip top>
-          <template #activator="{ on }">
-            <div v-on="on">
-              <VAvatar v-if="counterparty">
-                <VIcon v-if="counterparty.icon" :color="counterparty.color">
-                  {{ counterparty.icon }}
-                </VIcon>
-
-                <VImg
-                  v-else-if="counterparty.image"
-                  :src="`${imagePath}${counterparty.image}`"
-                  contain
-                />
+    <DefineImage>
+      <div>
+        <VAvatar v-if="counterparty" size="24">
+          <VIcon v-if="counterparty.icon" :color="counterparty.color">
+            {{ counterparty.icon }}
+          </VIcon>
```

Comment:
Hmmm damn this would require changes in the backend to migrate from mdi right?

Reply:
Yes it is, but we can do it later hahaha, it will not be hard to do